### ISSUE TITLE
linter: `id-length` - false positive on `in` generics

### ISSUE THREAD

**Issue by DetachHead** [oxc-project/oxc#23801]
### What version of Oxlint are you using?

1.71.0

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "categories": {
    "correctness": "off"
  },
  "plugins": ["react"],
  "rules": {
    "id-length": ["warn", { "checkGeneric": false }]
  }
}
```

### What happened?

```ts
type Foo<T> = { // no error (correct)
  [K in keyof T]: unknown; // error on K (false positive)
};
```
